extensions/deprecated/pom.xml

```diff
@@ -0,0 +1,61 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright (C) 2019 jsonwebtoken.io
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~     http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
```

Comment:
Shouldn't we create two modules? `jjwt-jackson-deprecated` and `jjwt-orgjson-deprecated` ?

It feels funny to me to have two different implementations in the same module.

Thoughts?

Reply:
That thought crossed my mind, but this module, is just a shim, the implementations are not contained within, and you still need to add one of the implementations to the classpath.
The backward-compatible dependencies would look something like:
- `jjwt-api`
- `jjwt-impl`
- `jjwt-deprecated`
- `jjwt-jackson`

This would allow for runtime compatibility, and keep the extra modules to a minimum.

I'm not against adding separate impl modules.  I could probably even do it as a `classifier` (by shading/renaming the bits at build time):
- `jjwt-jackson-0.11.0-deprecated.jar`
- `jjwt-orgjson-0.11.0-deprecated.jar`


🤔 I'm kinda digging this later option, as would keep everything in the same namespaces, and we wouldn't have a `*-deprecated` show up via a [search](https://search.maven.org/search?q=jjwt).

Thoughts?

Reply:
Ah, I remember now - because of the if/then/else logic in determining an implementation in `RuntimeClasspathSerializerLocator`, you only want to instantiate the implementation that exists in the appropriate `jjwt-<impl>.jar`.

If you want to use gson for JJWT but some random 3rd party dependency pulls in Jackson, the `RuntimeClasspathSerializerLocator` implementation will use Jackson as a priority, even if you wanted gson.

For that reason, I think the the separate impl modules might be cleaner/safer, albeit a bit more of a PITA for us since it's just a shim.  I like the `classifier` approach - that seems good!